webapp/src/core.ts

```diff
@@ -449,6 +456,67 @@ export function scrollIntoView(item: JQuery, margin = 0) {
     }
 }
 
+function giveFocusToFirstTag(e: JQueryKeyEventObject) {
+    let charCode = (typeof e.which == "number") ? e.which : e.keyCode
+    if (charCode === 9 && !e.shiftKey) {
+        e.preventDefault();
+        (e.data as HTMLElement).focus()
+    }
+}
+function giveFocusToLastTag(e: JQueryKeyEventObject) {
+    let charCode = (typeof e.which == "number") ? e.which : e.keyCode
+    if (charCode === 9 && e.shiftKey) {
+        e.preventDefault();
+        (e.data as HTMLElement).focus()
+    }
+};
+
+export function initializeFocusTabIndex(element: Element) {
+    if (element !== document.activeElement && element.contains(document.activeElement)) {
+        return
+    }
+
+    let firstFocused = element.getElementsByClassName("firstFocused")
+    let lastFocused = element.getElementsByClassName("lastFocused")
+    let firstTag: HTMLElement
+    let lastTag: HTMLElement
+    let jFirstTag: JQuery
+    let jLastTag: JQuery
+
+    if (firstFocused.length > 1) {
+        console.error("There must be maximum one element with the class firstFocused in the scope.")
+        return
+    } else if (firstFocused.length === 1) {
+        firstTag = firstFocused.item(0) as HTMLElement
+        jFirstTag = $(firstTag)
+        jFirstTag.off("keydown", giveFocusToLastTag)
```

Comment:
Refrain from using Jquery. 
el.removeEventListener(eventName, eventHandler);

Use this site: http://youmightnotneedjquery.com/